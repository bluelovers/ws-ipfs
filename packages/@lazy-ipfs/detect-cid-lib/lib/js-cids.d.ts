import JsCID from 'cids';
import { ITSPickExtra, ITSRequireAtLeastOne } from 'ts-type';
export declare const SymbolJsCID: unique symbol;
export type IRawJsCID = ITSRequireAtLeastOne<ITSPickExtra<JsCID, 'version' | 'multihash' | 'codec' | 'code', 'multibaseName'>, 'codec' | 'code'>;
export declare function isRawJsCIDLike<T extends IRawJsCID = IRawJsCID>(cid: T | unknown): cid is T;
export declare function isJsCID<T extends JsCID = JsCID>(cid: unknown): cid is T;
export declare function assertJsCID<T extends JsCID = JsCID>(cid: any): asserts cid is T;
export declare function toRawJsCID(cid: IRawJsCID): IRawJsCID;
export type IRawJsCIDFake<T extends IRawJsCID = IRawJsCID> = T & {
    [SymbolJsCID]?: true;
};
export declare function toRawJsCIDFake(cid: IRawJsCID): IRawJsCIDFake;
