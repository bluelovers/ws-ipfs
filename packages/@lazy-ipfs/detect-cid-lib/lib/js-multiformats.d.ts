import { CID as MultiformatsCID } from 'multiformats';
/**
 * @deprecated this is not exists
 * @see https://github.com/multiformats/js-multiformats/pull/109
 */
export declare const SymbolMultiformatsCID: unique symbol;
export interface IRawMultiformatsCID extends Pick<MultiformatsCID, 'version' | 'code' | 'multihash' | 'bytes'> {
}
export declare function isRawMultiformatsCIDLike<T extends IRawMultiformatsCID>(cid: T | unknown): cid is T;
export declare function isMultiformatsCID<T extends MultiformatsCID = MultiformatsCID>(cid: unknown): cid is T;
export declare function assertMultiformatsCID<T extends MultiformatsCID = MultiformatsCID>(cid: any): asserts cid is T;
