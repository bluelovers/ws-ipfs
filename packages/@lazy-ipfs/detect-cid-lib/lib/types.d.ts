import JsCID from 'cids';
import { CID as MultiformatsCID } from 'multiformats';
import { IRawJsCID } from './js-cids';
import { IRawMultiformatsCID } from './js-multiformats';
export declare type ICIDObject = JsCID | MultiformatsCID;
export declare type ICIDRawObject = IRawJsCID | IRawMultiformatsCID;
export declare type IRawCIDVersion = ICIDObject["version"];
export declare type ICIDValue = ICIDObject | string;
export declare type ICIDValueOrRaw = ICIDValue | ICIDRawObject;
export declare type I_CIDLike = Pick<ICIDObject, 'version' | 'multihash'>;
