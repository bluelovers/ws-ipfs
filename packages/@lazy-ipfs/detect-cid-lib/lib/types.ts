import JsCID from 'cids';
import { CID as MultiformatsCID } from 'multiformats';
import { IRawJsCID } from './js-cids';
import { IRawMultiformatsCID } from './js-multiformats';

export type ICIDObject = JsCID | MultiformatsCID;
export type ICIDRawObject = IRawJsCID | IRawMultiformatsCID;

export type IRawCIDVersion = ICIDObject["version"];
export type ICIDValue = ICIDObject | string;
export type ICIDValueOrRaw = ICIDValue | ICIDRawObject;

export type I_CIDLike = Pick<ICIDObject, 'version' | 'multihash'>
