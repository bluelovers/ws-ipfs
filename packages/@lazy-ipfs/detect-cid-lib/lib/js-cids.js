"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.toRawJsCIDFake = exports.toRawJsCID = exports.assertJsCID = exports.isJsCID = exports.isRawJsCIDLike = exports.SymbolJsCID = void 0;
const tslib_1 = require("tslib");
const util_1 = require("./util");
const err_code_1 = (0, tslib_1.__importDefault)(require("err-code"));
const multicodec_1 = require("multicodec");
exports.SymbolJsCID = Symbol.for("@ipld/js-cid/CID" /* js_cids */);
function isRawJsCIDLike(cid) {
    return (0, util_1._isCIDLike)(cid) && (typeof cid.codec === 'string' || typeof cid.code === 'number') && (0, util_1._isArrayLike)(cid.multihash);
}
exports.isRawJsCIDLike = isRawJsCIDLike;
function isJsCID(cid) {
    return ((cid === null || cid === void 0 ? void 0 : cid[exports.SymbolJsCID]) === true) && isRawJsCIDLike(cid) && typeof cid.toV1 === 'function';
}
exports.isJsCID = isJsCID;
function assertJsCID(cid) {
    if (!isJsCID(cid)) {
        throw (0, err_code_1.default)(new TypeError(`CID is not '${"@ipld/js-cid/CID" /* js_cids */}'`), {
            input: cid,
        });
    }
}
exports.assertJsCID = assertJsCID;
function toRawJsCID(cid) {
    let { version, codec, code, multihash } = cid;
    // @ts-ignore
    code !== null && code !== void 0 ? code : (code = (0, multicodec_1.getCodeFromName)(codec));
    // @ts-ignore
    codec !== null && codec !== void 0 ? codec : (codec = (0, multicodec_1.getNameFromCode)(code));
    return {
        version,
        codec,
        code,
        multihash,
    };
}
exports.toRawJsCID = toRawJsCID;
function toRawJsCIDFake(cid) {
    const value = toRawJsCID(cid);
    value[exports.SymbolJsCID] = true;
    return value;
}
exports.toRawJsCIDFake = toRawJsCIDFake;
//# sourceMappingURL=js-cids.js.map