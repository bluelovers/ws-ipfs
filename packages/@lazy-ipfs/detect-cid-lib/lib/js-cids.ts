import JsCID from 'cids';
import {
	EnumTypeofCID,
} from '../index';
import { _isCIDLike, _isArrayLike } from './util';
import { ITSPickExtra, ITSRequireAtLeastOne } from 'ts-type';
import err_code from 'err-code';
import { getCodeFromName, getNameFromCode } from 'multicodec';

export const SymbolJsCID = Symbol.for(EnumTypeofCID.js_cids);

export type IRawJsCID = ITSRequireAtLeastOne<ITSPickExtra<JsCID, 'version' | 'multihash' | 'codec' | 'code', 'multibaseName'>, 'codec' | 'code'>;

export function isRawJsCIDLike<T extends IRawJsCID = IRawJsCID>(cid: T | unknown): cid is T
{
	return _isCIDLike(cid) && (typeof cid.codec === 'string' || typeof cid.code === 'number') && _isArrayLike(cid.multihash)
}

export function isJsCID<T extends JsCID = JsCID>(cid: unknown): cid is T
{
	return (cid?.[SymbolJsCID] === true) && isRawJsCIDLike<JsCID>(cid) && typeof cid.toV1 === 'function'
}

export function assertJsCID<T extends JsCID = JsCID>(cid: any): asserts cid is T
{
	if (!isJsCID(cid))
	{
		throw err_code(new TypeError(`CID is not '${EnumTypeofCID.js_cids}'`), {
			input: cid,
		})
	}
}

export function toRawJsCID(cid: IRawJsCID): IRawJsCID
{
	let { version, codec, code, multihash } = cid;

	// @ts-ignore
	code ??= getCodeFromName(codec);
	// @ts-ignore
	codec ??= getNameFromCode(code);

	return {
		version,
		codec,
		code,
		multihash,
	}
}

export type IRawJsCIDFake<T extends IRawJsCID = IRawJsCID> = T
	& {
	[SymbolJsCID]?: true
}

export function toRawJsCIDFake(cid: IRawJsCID): IRawJsCIDFake
{
	const value: IRawJsCIDFake = toRawJsCID(cid)

	value[SymbolJsCID] = true;

	return value
}
