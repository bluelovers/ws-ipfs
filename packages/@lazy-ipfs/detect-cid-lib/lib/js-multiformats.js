"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.toRawMultiformatsCIDFake = exports.toRawMultiformatsCID = exports.assertMultiformatsCID = exports.isMultiformatsCID = exports.isRawMultiformatsCIDLike = exports.SymbolMultiformatsCID = void 0;
const tslib_1 = require("tslib");
const js_cids_1 = require("./js-cids");
const util_1 = require("./util");
const err_code_1 = tslib_1.__importDefault(require("err-code"));
/**
 * @deprecated this is not exists
 * @see https://github.com/multiformats/js-multiformats/pull/109
 */
exports.SymbolMultiformatsCID = Symbol.for("@ipld/js-multiformats/CID" /* EnumTypeofCID.multiformats_cid */);
function isRawMultiformatsCIDLike(cid) {
    return (0, util_1._isCIDLike)(cid) && typeof cid.code === 'number' && (0, util_1._isArrayLike)(cid.bytes) && !(0, util_1._isArrayLike)(cid.multihash);
}
exports.isRawMultiformatsCIDLike = isRawMultiformatsCIDLike;
function isMultiformatsCID(cid) {
    return !(0, js_cids_1.isJsCID)(cid) && isRawMultiformatsCIDLike(cid) && typeof cid.toV1 === 'function' && (cid === null || cid === void 0 ? void 0 : cid.asCID) === cid;
}
exports.isMultiformatsCID = isMultiformatsCID;
function assertMultiformatsCID(cid) {
    if (!isMultiformatsCID(cid)) {
        throw (0, err_code_1.default)(new TypeError(`CID is not '${"@ipld/js-multiformats/CID" /* EnumTypeofCID.multiformats_cid */}'`), {
            input: cid,
        });
    }
}
exports.assertMultiformatsCID = assertMultiformatsCID;
function toRawMultiformatsCID(cid) {
    const { version, code, multihash, bytes } = cid;
    const value = {
        version,
        code,
        multihash,
        bytes,
    };
    return value;
}
exports.toRawMultiformatsCID = toRawMultiformatsCID;
function toRawMultiformatsCIDFake(cid) {
    const value = toRawMultiformatsCID(cid);
    value.asCID = value;
    return value;
}
exports.toRawMultiformatsCIDFake = toRawMultiformatsCIDFake;
//# sourceMappingURL=js-multiformats.js.map