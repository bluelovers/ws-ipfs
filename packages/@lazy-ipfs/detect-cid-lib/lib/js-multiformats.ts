import { CID as MultiformatsCID } from 'multiformats';
import { EnumTypeofCID } from '../index';
import { isJsCID } from './js-cids';
import { _isCIDLike, _isArrayLike } from './util';
import err_code from 'err-code';

/**
 * @deprecated this is not exists
 * @see https://github.com/multiformats/js-multiformats/pull/109
 */
export const SymbolMultiformatsCID = Symbol.for(EnumTypeofCID.multiformats_cid);

export interface IRawMultiformatsCID extends Pick<MultiformatsCID, 'version' | 'code' | 'multihash' | 'bytes'>
{

}

export function isRawMultiformatsCIDLike<T extends IRawMultiformatsCID>(cid: T | unknown): cid is T
{
	return _isCIDLike(cid) && typeof cid.code === 'number' && _isArrayLike(cid.bytes) && !_isArrayLike(cid.multihash)
}

export function isMultiformatsCID<T extends MultiformatsCID = MultiformatsCID>(cid: unknown): cid is T
{
	return !isJsCID(cid) && isRawMultiformatsCIDLike<MultiformatsCID>(cid) && typeof cid.toV1 === 'function' && (cid as any)?.asCID === cid
}

export function assertMultiformatsCID<T extends MultiformatsCID = MultiformatsCID>(cid: any): asserts cid is T
{
	if (!isMultiformatsCID(cid))
	{
		throw err_code(new TypeError(`CID is not '${EnumTypeofCID.multiformats_cid}'`), {
			input: cid,
		})
	}
}

export function toRawMultiformatsCID(cid: IRawMultiformatsCID): IRawMultiformatsCID
{
	const { version, code, multihash, bytes } = cid

	const value: IRawMultiformatsCID = {
		version,
		code,
		multihash,
		bytes,
	}

	return value
}

export type IRawMultiformatsCIDFake<T extends IRawMultiformatsCID = IRawMultiformatsCID> = T
	& {
	asCID?: IRawMultiformatsCIDFake<T>;
}

export function toRawMultiformatsCIDFake(cid: IRawMultiformatsCID): IRawMultiformatsCIDFake
{
	const value: IRawMultiformatsCIDFake = toRawMultiformatsCID(cid)

	value.asCID = value;

	return value
}
