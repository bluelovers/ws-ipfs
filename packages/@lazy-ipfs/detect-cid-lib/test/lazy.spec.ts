import typeofCID, { EnumTypeofCID} from '../index';
import JsCID from 'cids';
import { CID as MultiformatsCID } from 'multiformats';
import { isMultiformatsCID } from '../lib/js-multiformats';
import { isJsCID } from '../lib/js-cids';

describe(`describe`, () =>
{

	test(EnumTypeofCID.js_cids, () =>
	{
		const cid = new JsCID('bafybeidw5mmxmimpuzo7wiwmhwvea2zalug5djderubacu2aqnneva6zwy');

		let actual = typeofCID(cid);
		let expected = EnumTypeofCID.js_cid;

		console.dir(cid)

		expect(cid.multihash).toMatchSnapshot();
		expect(cid).toMatchSnapshot();
		expect(actual).toStrictEqual(expected);
		expect(isJsCID(cid)).toBeTruthy();

	});

	test(EnumTypeofCID.multiformats_cid, () =>
	{
		const cid = MultiformatsCID.parse('bafybeidw5mmxmimpuzo7wiwmhwvea2zalug5djderubacu2aqnneva6zwy');

		let actual = typeofCID(cid);
		let expected = EnumTypeofCID.multiformats_cid;

		console.dir(cid)

		expect(cid.multihash).toMatchSnapshot();
		expect(cid).toMatchSnapshot();
		expect(actual).toStrictEqual(expected);
		expect(isMultiformatsCID(cid)).toBeTruthy();

	});

})
