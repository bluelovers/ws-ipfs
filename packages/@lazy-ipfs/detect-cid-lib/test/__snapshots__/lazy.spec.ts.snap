// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`describe @ipld/js-cid/CID 1`] = `
Uint8Array [
  18,
  32,
  118,
  235,
  25,
  118,
  33,
  143,
  166,
  93,
  251,
  34,
  204,
  61,
  170,
  64,
  107,
  32,
  93,
  13,
  209,
  164,
  100,
  141,
  2,
  1,
  83,
  64,
  131,
  90,
  74,
  131,
  217,
  182,
]
`;

exports[`describe @ipld/js-cid/CID 2`] = `
{
  "codec": "dag-pb",
  "hash": Uint8Array [
    18,
    32,
    118,
    235,
    25,
    118,
    33,
    143,
    166,
    93,
    251,
    34,
    204,
    61,
    170,
    64,
    107,
    32,
    93,
    13,
    209,
    164,
    100,
    141,
    2,
    1,
    83,
    64,
    131,
    90,
    74,
    131,
    217,
    182,
  ],
  "version": 1,
}
`;

exports[`describe @ipld/js-multiformats/CID 1`] = `
Digest {
  "bytes": Uint8Array [
    18,
    32,
    118,
    235,
    25,
    118,
    33,
    143,
    166,
    93,
    251,
    34,
    204,
    61,
    170,
    64,
    107,
    32,
    93,
    13,
    209,
    164,
    100,
    141,
    2,
    1,
    83,
    64,
    131,
    90,
    74,
    131,
    217,
    182,
  ],
  "code": 18,
  "digest": Uint8Array [
    118,
    235,
    25,
    118,
    33,
    143,
    166,
    93,
    251,
    34,
    204,
    61,
    170,
    64,
    107,
    32,
    93,
    13,
    209,
    164,
    100,
    141,
    2,
    1,
    83,
    64,
    131,
    90,
    74,
    131,
    217,
    182,
  ],
  "size": 32,
}
`;

exports[`describe @ipld/js-multiformats/CID 2`] = `
{
  "code": 112,
  "hash": Uint8Array [
    18,
    32,
    118,
    235,
    25,
    118,
    33,
    143,
    166,
    93,
    251,
    34,
    204,
    61,
    170,
    64,
    107,
    32,
    93,
    13,
    209,
    164,
    100,
    141,
    2,
    1,
    83,
    64,
    131,
    90,
    74,
    131,
    217,
    182,
  ],
  "version": 1,
}
`;
